### PR TITLE
User

### PR DESCRIPTION
**Pull request by junzai97**
User modules

### CODE REVIEW

File: src/routers/users.router.js [hunk 1]
```diff
@@ -1,17 +1,141 @@
...
+            toMysqlTimestampString(new Date()),
+            toMysqlTimestampString(new Date())
+        );
+    
+        const salt = await bcrypt.genSalt(5);
+        user.password = await bcrypt.hash(user.password, salt);
+        const result = await createUser(user);
+
+        const token = jwt.sign({id: result.insertId.toString()}, process.env.JWT_SECRET, {expiresIn: 86400})
+        res.status(200).send({auth: true, token});
```
Comment (kachun333) [inline src/routers/users.router.js]: can use `201` instead of 200?

File: src/routers/users.router.js [hunk 2]
```diff
@@ -1,17 +1,141 @@
...
+    const userDetailsDTO = req.body;
+    const invalidUserDetailsDTO = hasMissingKey(userDetailsDTO, new UserDetailsDTO());
+    if(invalidUserDetailsDTO){
+        res.status(400).send("Invalid request body")
+    }
+    try {
+        const savedStorageResult = await createStorage(
+            userDetailsDTO.base64String,
+            AttachmentType.USER
+        );
```
Comment (kachun333) [inline src/routers/users.router.js:81]: means every time a user updates his/her profile, s/he must change his/her profile image?
even if the profile image is the same, we will save the same image to firebase storage?

File: ddl_script/08012021_alter_user_table.sql [hunk 3]
```diff
@@ -0,0 +1 @@
+ALTER TABLE user DROP COLUMN firebase_uid;
```
Comment (kachun333) [inline ddl_script/08012021_alter_user_table.sql]: you forget to add `password` column

File: src/middleware/auth.middleware.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+const jwt = require('jsonwebtoken');
+const { getUserWithId } = require('../repository/users.repostitory')
+
+const auth = async (req, res, next) => {
+    try {
+        const token = req.header('Authorization').replace('Bearer ', '');
+        const decoded = jwt.verify(token, process.env.JWT_SECRET);
```
Comment (kachun333) [inline src/middleware/auth.middleware.js:7]: please update readme.md to remind developer to setup `process.env.JWT_SECRET` in their env file

File: src/routers/users.router.js [hunk 5]
```diff
@@ -1,17 +1,141 @@
...
+        
+        const user = {
+            id: req.user.id,
+            imageStorageId : savedStorageResult.id,
+            gender : userDetailsDTO.gender,
+            biography: userDetailsDTO.biography,
+            lastModifiedDate: toMysqlTimestampString(new Date()),
+        }
+
+        res
```
Comment (kachun333) [inline src/routers/users.router.js:93]: you forget to update the user in DB, and you str8 return `200` to user

File: src/middleware/auth.middleware.js (hunk 4 above)
Comment (junzai97) [reply src/middleware/auth.middleware.js:7]: Solved in  b09bae5

File: src/routers/users.router.js (hunk 2 above)
Comment (junzai97) [reply src/routers/users.router.js:81]: This API will only called by the user once, during the registration.
For update User info, another PUT API will be created, only there we will do the checking for the existence of base64String in req,body.

File: src/routers/users.router.js (hunk 1 above)
Comment (junzai97) [reply src/routers/users.router.js]: Solved in 007618a918bc240189032ba2189f1136c681a8f9

File: ddl_script/08012021_alter_user_table.sql (hunk 3 above)
Comment (junzai97) [reply ddl_script/08012021_alter_user_table.sql]: solved in 007618a918bc240189032ba2189f1136c681a8f9

File: src/routers/users.router.js (hunk 5 above)
Comment (junzai97) [reply src/routers/users.router.js:93]: solved in 007618a918bc240189032ba2189f1136c681a8f9

File: src/routers/users.router.js (hunk 2 above)
Comment (kachun333) [reply src/routers/users.router.js:81]: noted.

**Review by kachun333 — APPROVED** [review]

**Pull request merged**